### PR TITLE
Upgrade to TypeScript 3

### PR DESCRIPTION
**Pull request by damccorm**
Pretty straightforward, only changes were due to writeFileSync now apparently requiring an object. Setting a string to the encoding field does the same thing anyways though.

### CODE REVIEW

**Comment by damccorm** [comment]
I don't think that this should have a significant impact on how code is compiled to JavaScript, mostly it just lets us do more things (like #233)

**Comment by damccorm** [comment]
@bryanmacfarlane are you ok to move forward with this?

File: node/task.ts [hunk 1]
```diff
@@ -554,7 +554,12 @@ export interface FsOptions {
 }
 
 export function writeFile(file: string, data: string | Buffer, options?: string | FsOptions) {
-    fs.writeFileSync(file, data, options);
+    if(typeof(options) == 'string'){
```
Comment (stephenmichaelf) [inline node/task.ts]: ===?

**Comment by stephenmichaelf** [comment]
For some reason I can't approve. New UX? :)

**Pull request merged**